src/segment-loader.js

```diff
@@ -667,17 +675,15 @@ export default videojs.extend(videojs.EventTarget, {
                                                    this.sourceUpdater_.buffered());
 
     // Update segment meta-data (duration and end-point) based on timeline
-    let timelineUpdated = updateSegmentMetadata(playlist,
-                                                currentMediaIndex,
-                                                timelineUpdate);
+    timelineUpdated = updateSegmentMetadata(playlist, currentMediaIndex, timelineUpdate);
 
     // the last segment append must have been entirely in the
     // already buffered time ranges. adjust the timestamp offset to
     // fetch forward until we find a segment that adds to the
     // buffered time ranges and improves subsequent media index
     // calculations.
-//    if (!timelineUpdated) {
-//      this.expired_ -= segment.duration;
-//    }
+    if (!timelineUpdated) {
+      this.expired_ -= segment.duration;
```

Comment:
I'll comment it back out again for now then. I think I added it back to fix some tests, but I'll skip them for the time being.